packages/core/src/Serialization/TransactionBody/Utils.ts

```diff
@@ -59,3 +60,10 @@ export const multiAssetsToTokenMap = (
 
   return tokenMap;
 };
+
+/** Peeks for a tag indicating a `set` 6.258, and performs read if present */
+export const tryReadSet = (reader: CborReader): void => {
```

Comment:
just an idea, what if we create a new serializable class 'Set.ts', which takes in the constructor an array, and knows how to serialize/deserialize sets, and we can put this in the set itself, it can deserialize both arrays with and without the tag, and use the tag to serialize only if the feature flag is enabled (and we can place the feature flag within the set module)